### PR TITLE
Add title to all readme sections

### PR DESCRIPTION
**Pull request by NickNaso**
I provide the title to the sections in the readme file so they can be linked by external resources like slides other issues and slides.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,14 +21,14 @@ will be warned that it is experimental as follows:
 
 Node.js versions 8.12.0 and above provide N-API as a stable feature.
 
-**Branches**
+## **Branches**
```
Comment (legendecas) [inline README.md]:
```suggestion
## Branches
```

Then we don't need these asterisks right?

Comment (NickNaso) [reply README.md]: Yes I can remove the asterisks from every title because they are unnecessary. Do you agree?

Comment (legendecas) [reply README.md]: Definitely, yes.

**Review by mhdawson — APPROVED** [review]
LGTM

**Review by legendecas — APPROVED** [review]

**Pull request merged**